### ISSUE TITLE
Limited update rate

### ISSUE THREAD

**Issue by nebulusdesign** [sparkfunX/Qwiic_IR_Array_MLX90640#1]
I’ve attached my sensor to a teensy3.2, and I’m struggling to get anything more than 16Hz.
I’ve remarked out the processing code so it’s just pulling the data from the sensor.

Is this a limitation in the i2c bus, as I’ve got 32Hz working with Pimoroni’s Raspberry Pi C code.

Are there any microprocessor’s that will get close to the maximum 64Hz rate?

**Comment by nseidle** [comment]
tldr; **4Hz** is the best I can do

I've been playing with this. 

The I2C at 400kHz looks non-stop:
![image](https://user-images.githubusercontent.com/117102/43735917-293a0680-9979-11e8-9022-79245642eb32.png)

Luckily we only need to read the EEPROM parameters at 400kHz. After that we can go to 1MHz. Because of clock division, the Teensy 3.5 is 800kHz or 1.2MHz. Looks like only 800kHz is supported (I have not tried [i2c_t3](https://github.com/nox771/i2c_t3)).

At 800kHz we spend as much time calculating as we do I2C reading

![image](https://user-images.githubusercontent.com/117102/43736518-1b35c1d0-997b-11e8-919b-76ab87c243c5.png)

In the image above it's approximately 72ms to calculate the temps at 120MHz. 

If we remove all calculation and serial printing I can get the output up to 16Hz effective (setting the MLX90640 to 32Hz).  Of course this is kind of silly because we're not really doing anything.

Now maths. Teensy @ 400kHz I2C = (logic analyzer shows) 372kHz = 37,000 bytes per second. The MLX90640 requires 1668 bytes per frame so that's a max of 22 frames per second. So 32Hz is out of the picture unless we use faster I2C. Teensy does Serial over USB so I'm not sure what kind of a bottleneck it introduces. My gut says not much, if any. 

This is all to say that 4Hz (8Hz setting on the MLX90640) is the max usable data rate I can get reliably.

I'm going to close the issue but let me know if you (or anyone else) find more info.

**Comment by nseidle** [comment]
Here's the code I used for testing.

[Example4_MaxRefreshRate.zip](https://github.com/sparkfunX/Qwiic_IR_Array_MLX90640/files/2263999/Example4_MaxRefreshRate.zip)

**Comment by nseidle** [comment]
I pushed this example to the [firmware repo](https://github.com/sparkfun/SparkFun_MLX90640_Arduino_Example).
